src/coreclr/vm/interpexec.cpp

```diff
@@ -1200,6 +1201,136 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     UNREACHABLE();
                     break;
                 }
+                case INTOP_NEWARR:
+                {
+                    int32_t length = LOCAL_VAR(ip[2], int32_t);
+                    if (length < 0)
+                        assert(0); // Interpreter-TODO: Invalid array length
+
+                    CORINFO_CLASS_HANDLE arrayClsHnd = (CORINFO_CLASS_HANDLE)pMethod->pDataItems[ip[3]];
+                    void* pHelper = pMethod->pDataItems[ip[4]];
+
+                    HELPER_NEWARR helperFn = reinterpret_cast<HELPER_NEWARR>(pHelper);
+
+                    Object* arr = helperFn(arrayClsHnd, (intptr_t)length);
+                    SetObjectReferenceUnchecked(LOCAL_VAR_ADDR(ip[1], OBJECTREF), ObjectToOBJECTREF(arr));
```

Comment:
You can write the `OBJECTREF` directly to the local var. No need for write barriers.